nullaway/src/main/java/com/uber/nullaway/GenericsChecks.java

```diff
@@ -0,0 +1,142 @@
+    CodeAnnotationInfo codeAnnotationInfo = CodeAnnotationInfo.instance(state.context);
+    if (config.isJSpecifyMode()) {
+      // typeArguments used in the instantiated type (like for Foo<String,Integer>, this gets
+      // [String,Integer])
+      // if base type is unannotated do not check for generics
+      // temporary check to handle testMapComputeIfAbsent
+      if (codeAnnotationInfo.isSymbolUnannotated(type.tsym, config)) {
+        return;
+      }
+
+      com.sun.tools.javac.util.List<Type> typeArguments = type.getTypeArguments();
+      HashSet<Integer> nullableTypeArguments = new HashSet<Integer>();
+      int index = 0;
+      for (Type typArgument : typeArguments) {
+        com.sun.tools.javac.util.List<Attribute.TypeCompound> annotationMirrors =
+            typArgument.getAnnotationMirrors();
+        boolean hasNullableAnnotation =
+            Nullness.hasNullableAnnotation(annotationMirrors.stream(), config);
+        if (hasNullableAnnotation) {
+          nullableTypeArguments.add(index);
+        }
+        index++;
+      }
+
+      Type baseType = type.tsym.type;
+      com.sun.tools.javac.util.List<Type> baseTypeArguments = baseType.getTypeArguments();
+      index = 0;
+      for (Type baseTypeArg : baseTypeArguments) {
+
+        // if type argument at current index has @Nullable annotation base type argument at that
+        // index
+        // should also have a @Nullable annotation on its upper bound.
+        if (nullableTypeArguments.contains(index)) {
+
+          Type upperBound = baseTypeArg.getUpperBound();
+          com.sun.tools.javac.util.List<Attribute.TypeCompound> annotationMirrors =
+              upperBound.getAnnotationMirrors();
+          boolean hasNullableAnnotation =
+              Nullness.hasNullableAnnotation(annotationMirrors.stream(), config);
+          // if base type argument does not have @Nullable annotation then the instantiation is
+          // invalid
+          if (!hasNullableAnnotation) {
+            invalidInstantiationError(tree, state, analysis);
+          }
+        }
+        index++;
+      }
+      // Generics check for nested type parameters
+      for (Type typeArgument : typeArguments) {
+        if (typeArgument.getTypeArguments().length() > 0) {
+          checkInstantiatedType(typeArgument, state, tree, analysis, config);
+        }
+      }
+    }
+  }
+
+  /** Generics checks for parameterized typed trees * */
+  public static void checkInstantiationForParameterizedTypedTree(
+      ParameterizedTypeTree tree, VisitorState state, NullAway analysis, Config config) {
+    if (!config.isJSpecifyMode()) {
+      throw new IllegalStateException("should only check type instantiations in JSpecify mode");
+    }
+    List<? extends Tree> typeArguments = tree.getTypeArguments();
+    HashSet<Integer> nullableTypeArguments = new HashSet<Integer>();
+    JCTree.JCTypeApply baseTypeTree = (JCTree.JCTypeApply) tree;
```

Comment:
In the previous method we used `baseType` to refer to the generic type whose variables are being instantiated.  Here `baseTypeTree` means something different as it is the same as the `ParameterizedTypeTree`.  I would just get rid of this local variable as it is only used once